scapy/contrib/mqtt.py

```diff
@@ -0,0 +1,251 @@
+## This file is part of Scapy
+## See http://www.secdev.org/projects/scapy for more informations
+## Copyright (C) Santiago Hernandez Ramos <shramos@protonmail.com>
+## This program is published under GPLv2 license
+
+
+from scapy.packet import *
+from scapy.fields import *
+from scapy.layers.inet import TCP
+
+
+# CUSTOM FIELDS
+
+class VariableFieldLenField(FieldLenField):
+    def addfield(self, pkt, s, val):
+        val = self.i2m(pkt, val)
+        data = []
+        while val:
+            if val > 127:
+                data.append(val & 127)
+                val /= 127
+            else:
+                data.append(val)
+                lastoffset = len(data) - 1
+                data = "".join(chr(val | (0 if i == lastoffset else 128))
+                               for i, val in enumerate(data))
+                return s + data
+            if len(data) > 3:
+                raise Scapy_Exception("%s: malformed length field" %
+                                      self.__class__.__name__)
+    def getfield(self, pkt, s):
+        value = 0
+        for offset, curbyte in enumerate(s):
+            curbyte = ord(curbyte)
+            value += (curbyte & 127) * (128 ** offset)
+            if curbyte & 128 == 0:
+                return s[offset + 1:], value
+            if offset > 2:
+                raise Scapy_Exception("%s: malformed length field" %
+                                      self.__class__.__name__)
+
+
+
+# LAYERS
+            
+control_packet_types = {1:'CONNECT (1)',
```

Comment:
Do not add the value in the names: `1: 'CONNECT'` is enough. This also applies to other `{value: name}` dictionaries in this code.